RecoMTD/TrackExtender/plugins/TrackExtenderWithMTD.cc

```diff
@@ -358,38 +545,256 @@ void TrackExtenderWithMTDT<TrackCollection>::produce(edm::Event& ev, const edm::
+    double beta_p;
+    double dt_p;
+
+    double prob_pi;
+    double prob_k;
+    double prob_p;
+  };
+
   void find_hits_in_dets(const MTDTrackingDetSetVector& hits,
+                         const Trajectory& traj,
                          const DetLayer* layer,
                          const TrajectoryStateOnSurface& tsos,
+                         const double& vtxTime,
+                         const reco::BeamSpot& bs,
                          const Propagator* prop,
                          const MeasurementEstimator& theEstimator,
                          const TransientTrackingRecHitBuilder& hitbuilder,
-                         TransientTrackingRecHit::ConstRecHitContainer& output) {
+                         bool useVtxConstraint,
+                         std::set<MTDHitMatchingInfo>& out) {
+    TrajectoryStateClosestToBeamLine tscbl;
+    bool tscbl_status = getTrajectoryStateClosestToBeamLine(traj, bs, prop, tscbl);
+
+    if (!tscbl_status)
+      return;
+
+    GlobalVector p = tscbl.trackStateAtPCA().momentum();
+
+    double pathlength;
+    trackPathLength(traj, bs, prop, pathlength);
+
     pair<bool, TrajectoryStateOnSurface> comp = layer->compatible(tsos, *prop, theEstimator);
     if (comp.first) {
       vector<DetLayer::DetWithState> compDets = layer->compatibleDets(tsos, *prop, theEstimator);
       if (!compDets.empty()) {
-        MTDTrackingRecHit* best = nullptr;
-        double best_chi2 = std::numeric_limits<double>::max();
         for (const auto& detWithState : compDets) {
           auto range = hits.equal_range(detWithState.first->geographicalId(), cmp_for_detset);
           for (auto detitr = range.first; detitr != range.second; ++detitr) {
             for (auto itr = detitr->begin(); itr != detitr->end(); ++itr) {
               auto est = theEstimator.estimate(detWithState.second, *itr);
-              if (est.first && est.second < best_chi2) {  // just take the best chi2
-                best = &(*itr);
-                best_chi2 = est.second;
-              }
+              auto pl = prop->propagateWithPath(tsos, detWithState.second.surface());
+
+              if (!est.first || std::abs(pl.second) == 0.)
+                continue;
+
+              double tot_pl = pathlength + std::abs(pl.second);  //
+              double t_vtx = useVtxConstraint ? vtxTime : 0.;
+
+              constexpr double vtx_res = 0.008;
+              constexpr double beamSpot_res = 0.180;
+              double t_vtx_err = useVtxConstraint ? vtx_res : beamSpot_res;  //should use beam spot in the future
+
+              constexpr double t_res_manual = 0.035;
+              TrackTofPidInfo tof(p.mag2(),
+                                  tot_pl,
+                                  itr->time(),
+                                  t_res_manual,
+                                  t_vtx,
+                                  t_vtx_err,
+                                  false);  //put hit error by hand for the moment
```

Comment:
it looks like the end-line comment made clang-format break up this call into many lines.

does the comment refer to the ```t_res_manual``` or to the full call?
For the former case, better make it on the line of the argument.
For the latter, perhaps move above the line ```TrackTofPidInfo tof(```